.github/workflows/deploy-pnpm.yaml

```diff
@@ -0,0 +1,48 @@
+# Workflow intended to properly build the GitHub Pages Site. Using PNPM 
+
+name: Deployment Docs - PNPM 
+
+on:
+  # Trigger deployment on every push to main branch.
+  push:
+    branches: [main]
+  # Trigger deployment manually.
+  workflow_dispatch:
+    
+jobs:
+  deploy:
+    runs-on: ubuntu-latest 
+    
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Fetch all commits to get last updated time or other git log info 
+          fetch-depth: 0
```

Comment:
I'm using this in many of my projects, but you should be aware of its downsides – specifying a fetch-depth of 0 won't let anyone re-run an older commit. Probably not relevant for a documentation repo.